pkg/cmd/workflow.go

```diff
@@ -0,0 +1,364 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/puppetlabs/relay/pkg/errors"
+	"github.com/puppetlabs/relay/pkg/format"
+	"github.com/puppetlabs/relay/pkg/model"
+	"github.com/puppetlabs/relay/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+func newWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "workflow",
+		Short: "Manage your relay workflows",
+		Args:  cobra.MinimumNArgs(1),
+	}
+
+	cmd.AddCommand(newAddWorkflowCommand())
+	cmd.AddCommand(newReplaceWorkflowCommand())
+	cmd.AddCommand(newDeleteWorkflowCommand())
+	cmd.AddCommand(newRunWorkflowCommand())
+	cmd.AddCommand(newListWorkflowsCommand())
+	cmd.AddCommand(newDownloadWorkflowCommand())
+
+	return cmd
+}
+
+func doAddWorkflow(cmd *cobra.Command, args []string) error {
+	file, ferr := readFile(cmd)
+
+	if ferr != nil {
+		return ferr
+	}
+
+	workflowName, nerr := getWorkflowName(args)
+
+	if nerr != nil {
+		return nerr
+	}
+
+	Dialog.Progress("Creating your workflow...")
+
+	workflow, cwerr := Client.CreateWorkflow(workflowName)
+
+	if cwerr != nil {
+		return cwerr
+	}
+
+	revision, rerr := Client.CreateRevision(workflow.Workflow.Name, file)
+
+	if rerr != nil {
+
+		// attempt to revert creation of workflow record
+		Client.DeleteWorkflow(workflow.Workflow.Name)
+
+		return rerr
+	}
+
+	wr := model.NewWorkflowRevision(workflow.Workflow, revision.Revision)
+
+	wr.Output(Config)
+
+	Dialog.Infof(`Successfully created workflow %v
+			
+View more information or update workflow settings at %v`,
+		workflow.Workflow.Name,
+		format.GuiLink(Config, "/workflow/%v", workflow.Workflow.Name),
```

Comment:
this url isn't correct, the path needs to be `/workflows/`

Reply:
Good catch! Just patched this.